packages/core/src/routers/__tests__/StackRouter.test.js

```diff
@@ -2063,4 +2060,46 @@ describe('StackRouter', () => {
     expect(activeState4.routeName).toEqual('NestedStack');
     expect(activeState4.routes[activeState4.index].routeName).toEqual('Bar');
   });
+
+  it('should replace existing params when set params on existing state', () => {
+    const Screen = () => <div />;
+
+    const router = StackRouter({
+      Foo: {
+        screen: Screen,
+        params: { a: 1 },
+      },
+    });
+    const state = router.getStateForAction({ type: NavigationActions.INIT });
+    expect(state).toEqual({
+      index: 0,
+      isTransitioning: false,
+      key: 'StackRouterRoot',
+      routes: [{ routeName: 'Foo', key: 'id-0', params: { a: 1 } }],
+    });
+
+    const newState = router.getStateForAction(
+      NavigationActions.setParams({ key: 'id-0', params: { b: 2 } }),
+      state
+    );
+
+    expect(newState.routes[newState.index].params).toEqual({ b: 2 });
+  });
+
+  it('should merge params when setting params during init', () => {
```

Comment:
Is this the expected behavior? Let me know what should happen when getting the initial state from a set-params action. Right now, the current behavior is to merge so that's why I've written the test like this, but let me know if I need to change that too.